### PR TITLE
Pin matplotlib for now

### LINKED ISSUE #28045: Pin matplotlib-base to 3.8 (or something older than 3.9)
### Motivation

A recent addition to matplotlib-base is calling `system_profiler`. Something that shouldn't be allowed to run on our build machines.

### Design

Pin matplotlib-base to a version before matplotlib began calling system_profiler. There is a fix moving through Conda-Forge that addresses this issue: https://github.com/matplotlib/matplotlib/issues/28485

### Impact

CI will continue to progress

### PR DESCRIPTION
**Pull request by milljm**
Temporarily pin an older matplotlib, until https://github.com/matplotlib/matplotlib/pull/28498
is resolved.

Closes https://github.com/idaholab/moose/issues/28045

### CODE REVIEW

**Review by loganharbour — APPROVED** [review]

File: conda/peacock/meta.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ requirements:
     - numpy
     - pandas
     - pyqt
-    - matplotlib
+    - matplotlib <3.9
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply conda/peacock/meta.yaml:28]: (also 👋🏻 from another national lab (BNL))

File: conda/tools/meta.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ requirements:
     - livereload
     - lxml
     - mako
-    - matplotlib-base
+    - matplotlib-base <3.9
```
Comment (tacaswell) [inline conda/tools/meta.yaml:44]:
```suggestion
    - matplotlib-base !=3.9.0
```

File: conda/peacock/meta.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (loganharbour) [reply conda/peacock/meta.yaml:28]: > (also 👋🏻 from another national lab (BNL))

Hah - thanks for the checkup 👋🏼

[the remark this answers is not in the document]
Comment (milljm) [reply conda/peacock/meta.yaml]: Yes thank you for this!

**Comment by milljm** [comment]
There is another larger PR in progress we would rather include this temporary fix in.

Closes this in favor of: #26718

**Pull request closed without merging**